hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/viewmodel/BlockType.java

```
@@ -16,23 +16,49 @@
 
 package com.hedera.mirror.web3.viewmodel;
 
+import java.util.Arrays;
 import org.apache.commons.lang3.StringUtils;
 
 public record BlockType(String name, long number) {
 
-    public static final BlockType EARLIEST = new BlockType("earliest", 0L);
-    public static final BlockType LATEST = new BlockType("latest", Long.MAX_VALUE);
-    public static final BlockType PENDING = new BlockType("pending", Long.MAX_VALUE);
+    private enum BlockTypeName {
+        EARLIEST("earliest"),
+        LATEST("latest"),
+        PENDING("pending"),
+        SAFE("safe"),
+        FINALIZED("finalized");
+
+        private final String name;
+
+        BlockTypeName(final String name) {
+            this.name = name;
+        }
+    }
+
     private static final String HEX_PREFIX = "0x";
+    public static final BlockType LATEST = new BlockType(BlockTypeName.LATEST.name, Long.MAX_VALUE);
+
+    private static final BlockTypeName[] UNSUPPORTED_TYPES = {
+        BlockTypeName.PENDING, BlockTypeName.SAFE, BlockTypeName.FINALIZED
+    };
 
     public static BlockType of(final String value) {
+        if (StringUtils.isEmpty(value)) {
+            return LATEST;
+        }
 
-        if (StringUtils.isEmpty(value) || BlockType.LATEST.name().equalsIgnoreCase(value)) {
-            return BlockType.LATEST;
-        } else if (BlockType.EARLIEST.name().equalsIgnoreCase(value)) {
-            return BlockType.EARLIEST;
-        } else if (BlockType.PENDING.name().equalsIgnoreCase(value)) {
-            return BlockType.PENDING;
+        try {
+            final BlockTypeName blockTypeName = BlockTypeName.valueOf(value.toUpperCase());
+            switch (blockTypeName) {
+                case EARLIEST -> {
+                    return new BlockType(BlockTypeName.EARLIEST.name, 0L);
+                }
+                case LATEST, PENDING, SAFE, FINALIZED -> {
```

We should just throw an exception if pending, safe or finalized.

---

Please check my other comment.